Documentation for content media type and content encoding (#234)
SAP/abap-file-formats-tools: README.md

Pull the source code with [abapGit](https://github.com/abapGit/abapGit/) into your ABAP system.
The latest abapGit build can be downloaded [here](https://raw.githubusercontent.com/abapGit/build/main/zabapgit_standalone.prog.abap).

The development of the ABAP file formats tools takes place in the `Z` namespace.
Coding policies are stated and verified by [abaplint](https://github.com/marketplace/abaplint), see configuration file [here](abaplint.jsonc), that run on every pull request.

> ## Usage

Reviewer: ```suggestion
## How To Use
```

> ## How To Use

Reviewer 2: ```suggestion
## How to Use
```

This repository mainly consists of two programs:
* `z_generate_json_schema` generates JSON Schemas (minimal requirement is v7.54) from ABAP types

> * `z_generate_repo` generates files and folder hierarchy for upload and maintain the [abap-file-formats](https://github.com/SAP/abap-file-formats) repository (requires ABAP file format framework)

Reviewer 2: ```suggestion
* `z_generate_repo` generates files and folder hierarchy for uploading and maintaining the [abap-file-formats](https://github.com/SAP/abap-file-formats) repository (requires ABAP file format framework)
```

To add more information to the JSON Schema than that provided by the ABAP type, ABAP Doc is parsed.

> Different possibilities on adding more information via annotations are listed [here](https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#writing-JSON-schema-with-ABAP-types).

Reviewer 2: ```suggestion
Different possibilities for adding more information via annotations are listed [here](https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#writing-JSON-schema-with-ABAP-types).
```

> Additionally to the AFF-related annotation, one can also use the following:

Reviewer: ```suggestion
Additionally to the ABAP-file-formats-related annotations, one can also use the following:
```

### Content Media Type
 To specify the content media type of the content of a string field, the annotation
 ```abap
 "! $contentMediaType 'mediaType'
 ```
 followed by the media type surrounded by single quotation marks is used.

>  This annotation can only be used for character like ABAP types, i.e., types that are mapped to JSON type `string`.

Reviewer 2: ```suggestion
 This annotation can only be used for characters like ABAP types, i.e., types that are mapped to JSON type `string`.
```

>  ### Content Encoding

Reviewer: I think we should remove the space in the beginning:

```suggestion
### Content Encoding
```

>  The encoding used to store the content of a string field can be specifed with the annotation

Reviewer 2: ```suggestion
 The encoding used to store the content of a string field can be specified with the annotation
```

 ```abap
 "! $contentEncoding 'encoding'
 ```
 followed by the encoding surrounded by single quotation marks.
 Possible values for the encoding are `7bit`, `8bit`, `binary`, `quoted-printable`, `base16`, `base32` and `base64`.

>  This annotation can also only be used for character like ABAP types, i.e., types that are mapped to JSON type `string`.

Reviewer 2: ```suggestion
 This annotation can also only be used for characters like ABAP types, i.e., types that are mapped to JSON type `string`.
```
